core/settings.py

```
@@ -24,10 +24,12 @@
 # See https://docs.djangoproject.com/en/6.0/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-try:
-    SECRET_KEY = os.environ['SECRET_KEY']
-except KeyError:
-    raise ImproperlyConfigured('SECRET_KEY environment variable is required')
+SECRET_KEY = os.environ.get('SECRET_KEY')
+if not SECRET_KEY:
+    if os.environ.get('VERCEL_ENV') == 'production':
+        raise ImproperlyConfigured('SECRET_KEY environment variable is required in production')
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)